pkg/util/manifests.go

```diff
@@ -0,0 +1,83 @@
+package util
+
+import (
+	"fmt"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+)
+
+// AddEnvVarToDeployment adds an environment variable to the given deployment containers for which the filter function returns true
+func AddEnvVarToDeployment(d *appsv1.Deployment, e v1.EnvVar, appendIfExists bool, filter func(c v1.Container) bool) {
+	ctrs := make([]v1.Container, 0)
+	for _, c := range d.Spec.Template.Spec.Containers {
+		if !filter(c) {
+			ctrs = append(ctrs, c)
+			continue
+		}
+		found := false
+		vars := make([]v1.EnvVar, 0)
+		for _, ce := range c.Env {
+			if ce.Name == e.Name {
+				found = true
+				if appendIfExists {
+					ce.Value = fmt.Sprintf("%s%s", ce.Value, e.Value)
+				} else {
+					ce.Value = e.Value
+				}
+			}
+			vars = append(vars, ce)
+		}
+		if !found {
+			vars = append(vars, e)
+		}
+		c.Env = vars
+		ctrs = append(ctrs, c)
+	}
+	d.Spec.Template.Spec.Containers = ctrs
+}
+
+// AddVolumeMountAsSecret creates a new volume and volume mount for the given secret and adds it to all containers for which the filter function returns true
+func AddVolumeMountAsSecret(d *appsv1.Deployment, s *v1.Secret, mountPath string, filter func(c v1.Container) bool) {
+	addSecretVolumeIfNeeded(s.Name, d)
+	addSecretMountIfNeeded(s.Name, mountPath, d, filter)
+}
+
+func addSecretVolumeIfNeeded(secretName string, d *appsv1.Deployment) {
+	vols := make([]v1.Volume, 0)
+	for _, vi := range d.Spec.Template.Spec.Volumes {
+		vols = append(vols, vi)
+		if vi.Name == secretName {
+			return
+		}
+	}
+	vols = append(vols, v1.Volume{
+		Name: secretName,
+		VolumeSource: v1.VolumeSource{
+			Secret: &v1.SecretVolumeSource{
+				SecretName: secretName,
+			},
+		},
+	})
+	d.Spec.Template.Spec.Volumes = vols
+}
+
+func addSecretMountIfNeeded(volumeName, mountPath string, d *appsv1.Deployment, filter func(c v1.Container) bool) {
+	ctrs := make([]v1.Container, 0)
+	for _, c := range d.Spec.Template.Spec.Containers {
+		if !filter(c) {
+			ctrs = append(ctrs, c)
+			continue
+		}
+		added := false
+		for _, vm := range c.VolumeMounts {
+			if vm.Name == volumeName && vm.MountPath == mountPath {
```

Comment:
Shouldn't it be an error to have two volumes mounted to the same path with different volume name? Either return the volume name to be use or allow the method to error.

Reply:
You're right, I'll handle that as an error, maybe there's a configmap already mounted in the same path where we want to mount a secret

Reply:
What if you just want to update the secret? I believe we already have similar code.

Reply:
If the secret has the same name and mount path this method would be noop, but if it has a different name then we need to delete it to add the new one instead. What about a bool param to indicate if we should replace any mounted volume?
```
func AddVolumeMountAsSecret(d *appsv1.Deployment, s *v1.Secret, mountPath string, replace bool, filter func(c v1.Container) bool) error {
}
```

Reply:
I wouldn't delete an existing secret implicitly like that. If we pass an existing volume name with a different mount path -> fine, we can mount twice (I believe?). If we mount two volumes to the mount path, it might actually work if the keys are specified and don't overlap (not sure either!) but it's probably not what was intended.

Maybe this is not a good candidate for a util function after all, and all we need is a `getVolumeMount(containername string, d *appsv1.Deployment) []v1.VolumeMount`, it's not worth it IMO.

Reply:
I see adding volumes as a common operation (we do that for k8s secrets and maybe also dynamic accounts). Maybe it's difficult to accomodate for a common enough method, I'll just leave the env var util method.